### PR TITLE
refactor(models): migrate downstream consumers to typed NemoClient

### PR DESCRIPTION
**Pull request by maxdubrinsky**
## Summary
Migrates the downstream model consumers off the Stainless-generated `sdk.inference.*` / `sdk.models.*` CRUD surface onto the typed `AsyncModelsClient` from the foundation landed in #993. This is the "downstream consumers" half of AIRCORE-876; the core `services/core/models` controllers are migrated in a separate sibling PR.

## Related Issue
Linear: AIRCORE-876 (Migrate Models service to NemoClient typed HTTP client), parent epic AIRCORE-827.

## Changes
- `services/platform-seed`: `seed_model_provider` now routes through `client_from_platform(sdk, AsyncModelsClient)` + `models.create_provider(..., body=CreateModelProviderRequest(...))` instead of `sdk.inference.providers.create(...)`.
- `services/core/entities` `WorkspaceCleanup._cleanup_deployments`: routes deployment listing/deletion through `AsyncModelsClient.list_deployments` / `delete_deployment` (preserving the `HeartbeatMixin` heartbeat calls recently added on main).
- `packages/nmp_customization_common` (model entity run task + platform client) and `packages/data_designer_nemo` model provider: switch to the typed models client.
- `services/automodel`, `services/rl`, `services/unsloth` job compilers: adopt the typed client.
- `services/guardrails` model routing / config utils: adopt the typed client.
- Adds `test_models_client_migration.py` focused tests for `data_designer_nemo`, `nmp_customization_common`, and `automodel`; updates existing suites (`test_workspace_cleanup`, `test_platform_seed_runner`, `test_model_routing`, `test_compiler`, `test_model_entity`, etc.) to the typed client mocking style.

The branch was rebased onto `origin/main` so the typed-client foundation comes from #993; the divergent in-branch foundation copy was dropped.

## Type of Change
- [x] Code change (feature, bug fix, or refactor)

## Quality Gates
- [x] Tests added or updated for changed behavior
- [ ] Documentation updated for user-visible behavior
- [x] Documentation not applicable — justification: internal client migration, no user-visible behavior change.

## Verification
- [x] Pull request title follows the repository's Conventional Commit format
- [x] Every commit includes an appropriate `Signed-off-by:` trailer
- [x] `uv run pre-commit run -a` passes, or any blocked checks are identified below
- [x] Targeted tests pass, or tests are marked not applicable above
- [x] No secrets, API keys, or credentials are included

Targeted validation:
- `uv run ruff check <changed .py files>` — All checks passed.
- `uv run --frozen pytest -q <affected test files>` — 101 passed in 0.45s (migration tests + updated existing suites for data_designer_nemo, nmp_customization_common, automodel, core/entities workspace_cleanup, guardrails, platform-seed, rl, unsloth).
- `uv run --frozen ty check <changed packages>` — 300 diagnostics, all pre-existing on `origin/main` (308 on main for the same set); the migration introduces no new type errors and removes 8 by replacing untyped SDK dict access with typed `AsyncModelsClient` calls.
- `uv run pre-commit run --files <changed files>` — ruff, ruff format, ty, copyright headers, merge-conflict checks all Passed.
- DCO audit: single commit `af4f86533`, `Signed-off-by: Max Dubrinsky <mdubrinsky@nvidia.com>` matches author email.

## Summary by CodeRabbit

* **Improvements**
  * Updated model, provider, deployment, and workspace operations to use current platform APIs.
  * Improved handling of transport, permission, not-found, and conflict errors.
  * Deployment configuration and model routing now use validated requests and responses.
  * Workspace cleanup continues to list and remove deployments reliably.
* **Tests**
  * Expanded coverage for model access, deployment configuration, provider creation, routing, cleanup, and deployment edge cases.

### CODE REVIEW

File: plugins/nemo-unsloth/tests/test_jobs.py [hunk 1]
```diff
@@ -39,21 +41,26 @@ def _input_dict(**overrides: Any) -> dict[str, Any]:
...
         fileset="base-fs",
         trust_remote_code=False,
     )
-    return SimpleNamespace(
+    sdk = SimpleNamespace(
         models=SimpleNamespace(retrieve=AsyncMock(return_value=me)),
         files=SimpleNamespace(
             filesets=SimpleNamespace(retrieve=AsyncMock(return_value=SimpleNamespace())),
         ),
     )
```
[the remark this answers is not in the document]
Comment (maxdubrinsky) [reply plugins/nemo-unsloth/tests/test_jobs.py:62]: correct but not useful.

File: services/core/entities/src/nmp/core/entities/controllers/workspace_cleanup.py [hunk 2]
```diff
@@ -161,13 +162,14 @@ async def _cleanup_jobs(self, workspace: Workspace) -> None:
...
+            deployments = [deployment async for deployment in deployments_response.items()]
 
             for deployment in deployments:
                 try:
                     logger.info(f"Deleting deployment: {deployment.name}")
-                    await self._nmp_sdk.inference.deployments.delete(
+                    await models_client.delete_deployment(
                         name=deployment.name,
                         workspace=workspace.name,
                     )
```
[the remark this answers is not in the document]
Comment (maxdubrinsky) [reply services/core/entities/src/nmp/core/entities/controllers/workspace_cleanup.py:175]: rejected

**Review by marcusds — APPROVED** [review]
